components/motor/gpio/controlled.go

```diff
@@ -324,6 +339,14 @@ func (cm *controlledMotor) GoFor(ctx context.Context, rpm, revolutions float64,
 		return err
 	}
 
+	// if loop is tuning, return an error
+	// if loop has been tuned but the values haven't been added to the config, error with tuned values
+	if cm.loop != nil && cm.loop.GetTuning(ctx) {
+		return control.TuningInProgressErr(cm.Name().ShortName())
+	} else if cm.configPIDVals[0].NeedsAutoTuning() && !(*cm.tunedVals)[0].NeedsAutoTuning() {
+		return control.TunedPIDErr(cm.Name().ShortName(), *cm.tunedVals)
+	}
+
```

Comment:
I'll try and add a control helper

Reply:
FWIW I think these should be helper in the scope of this package, not the control package. They care about things passed into the driver code at the driver level, it just makes it a little bit less repeated, so if you ever need to change its check in this package, you don't end up with a mismatch within the driver code itsel.f

Reply:
I think the controls package could own the helper, since the controls package kinda owns the tunedVals in the first place.   That was something I wanted to setup when proposing the refactors.  fine with leaving it in the driver for this pr

Reply:
we can incrementally add to the scope to the eventual PR, don't want to go too DRY in case we limit ourselves while addressing a driver functionality request, which is this ticket.